ingest-app/src/cmr/ingest/services/jobs_subscriptions.clj

```diff
@@ -0,0 +1,199 @@
+                start
+                (t/minus end (t/seconds amount-in-sec)))]
+    (str begin "," end)))
+
+(defn- search-gran-refs-by-collection-id
+  [context params1 params2 sub-id]
+  (try
+    (let [gran-refs1 (search/find-granule-references context params1)
+          gran-refs2 (search/find-granule-references context params2)]
+      (distinct (concat gran-refs1 gran-refs2)))
+    (catch Exception e
+      (error "Exception caught processing subscription" sub-id " searching with params "
+             (dissoc params1 :token) (dissoc params2 :token) "\n\n" (.getMessage e) "\n\n" e)
+      [])))
+
+(defn- process-subscriptions
+  "Process each subscription in subscriptions into tuple for testing purposes and to use as
+   the input when sending the subscription emails."
+  [context subscriptions]
+  (for [raw-subscription subscriptions
+        :let [time-constraint (subscription->time-constraint
+                               raw-subscription
+                               (t/now)
+                               (email-subscription-processing-lookback))
+              subscription (add-updated-since raw-subscription time-constraint)
+              subscriber-id (get-in subscription [:extra-fields :subscriber-id])
+              sub-id (get subscription :concept-id)
+              coll-id (get-in subscription [:extra-fields :collection-concept-id])
+              query-string (-> (:metadata subscription)
+                               (json/decode true)
+                               :Query)
+              query-params (create-query-params query-string)
+              params1 (merge {:created-at time-constraint
+                              :collection-concept-id coll-id
+                              :token (config/echo-system-token)}
+                             query-params)
+              params2 (merge {:revision-date time-constraint
+                              :collection-concept-id coll-id
+                              :token (config/echo-system-token)}
+                             query-params)
+              gran-refs (search-gran-refs-by-collection-id context params1 params2 sub-id)
+              subscriber-filtered-gran-refs (filter-gran-refs-by-subscriber-id context gran-refs subscriber-id)]]
+    (do
+      (send-update-subscription-notification-time! context sub-id)
+      [sub-id subscriber-filtered-gran-refs subscriber-id subscription])))
+
+(defn- send-subscription-emails
+  "Takes processed processed subscription tuples and sends out emails if applicable."
+  [context subscriber-filtered-gran-refs-list]
+  (doseq [subscriber-filtered-gran-refs-tuple subscriber-filtered-gran-refs-list
+          :let [[sub-id subscriber-filtered-gran-refs subscriber-id subscription] subscriber-filtered-gran-refs-tuple]]
+    (when (seq subscriber-filtered-gran-refs)
+      (let [gran-ref-locations (map :location subscriber-filtered-gran-refs)
+            email-address (urs/get-user-email context subscriber-id)
```

Comment:
At a certain point, we just have to trust that URS is giving us good data back, but as I mention in my comment below - should there be some method by which we return to an admin if the subscription they set up fails? Could happen due to misspelled username, for instance. Maybe we should check that a user exists in EDL at ingest time?

Reply:
well, email accounts could dead, URS accounts could be deactivated. I suggest we investigate account deactivations from URS as we know it will happen to everyone at some point.

Reply:
https://bugs.earthdata.nasa.gov/browse/CMR-7196